modules/git/git.go

```diff
@@ -242,6 +242,13 @@ func syncGitConfig() (err error) {
 		}
 	}
 
+	if CheckGitVersionAtLeast("1.8.2") == nil {
+		// The `histogram` algorithm itself was added in 1.7.7, but `diff.algorithm` was supported in Git 1.8.2.
+		if err := configSet("diff.algorithm", "histogram"); err != nil {
+			return err
+		}
+	}
```

Comment:
```suggestion
	if err := configSet("diff.algorithm", "histogram"); err != nil {
		return err
	}
```
The minimal required git version is already 2.0

However, I can't LGTM yet as I don't know what the side effects are.
Does it need more memory?
More CPU time?
Produce worse outputs?
Or why isn't it the default?

Reply:
Much of these questions are answered in this post: https://stackoverflow.com/questions/32365271/whats-the-difference-between-git-diff-patience-and-git-diff-histogram

> Does it need more memory?

There has been a short history of bad memory usage that has since been fixed:

>> Note: "git diff --histogram" had a bad memory usage pattern, which has been rearranged to reduce the peak usage, with Git 2.19 (Q3 2018).

>> Note: the patience and histogram algorithms had memory leaks, fixed with Git 2.36 (Q2 2022)

I can bump the version requirement if that's desirable.

> More CPU time?

It appears to be faster than both Myers and Patience.

>> [Commit 8c912ee](https://github.com/git/git/commit/8c912eea94a2138e8bc608f7c390eb0b313effb0) actually introduced --histogram to diff:
>> Port JGit's HistogramDiff algorithm over to C. Rough numbers (TODO) show that it is faster than its --patience cousin, as well as the default Meyers algorithm.

>> 8c912ee (teach --histogram to diff, 2011-07-12) claimed histogram diff was faster than both Myers and patience.

> Produce worse outputs?

Based on the [research article](https://link.springer.com/article/10.1007/s10664-019-09772-z) I posted in the ticket, it consistently produces better outputs than Myers. I'm broadly summarizing, but this is measured by finding over x2 more bugs in aggregate in the research.

> Why isn't it the default?

... uhh, idk. It's newer? :man_shrugging: